src/social_auth/utils.py

```diff
@@ -133,7 +138,10 @@ def ctype_to_model(val):
         ctype = ContentType.objects.get_for_id(val["ctype"])
         ModelClass = ctype.model_class()
         assert ModelClass is not None
-        val = ModelClass.objects.get(pk=val["pk"])
+        return ModelClass.objects.get(pk=val["pk"])
+
+    if isinstance(val, dict) and "username" in val and "name" in val:
+        return RpcUser.parse_obj(val)
```

Comment:
Could this receive a `User` object safely, or are we concrete in that only `RpcUser`s make it to this point. If it could be a mix of either, you may want to add a key to the dict that designates it as an `Rpc` instance to parse.

Reply:
Actually, nevermind, it looks like the Model objects turn objects from their primary keys, they aren't serialized, my bad.